### PR TITLE
PWGHF: Add TPC/TOF nSig hist in Xic task

### PR DESCRIPTION
**Pull request by phymanshu**
- These modifications in `taskXic.cxx` will enable to produce TH2F for TPC/TOF nSig vs P for Xic prongs
- A small tree is also added in order to see cut variation effects. This option, to write or not a TTree, is configurable.

### CODE REVIEW

**Review by lvermunt — CHANGES_REQUESTED** [review]
Hi @phymanshu,

Thanks. Few quick comments from a first look:
 - I'm not sure if this is a convention (@vkucera can probably better comment), but we typically use tasks like `PWGHF/TableProducer/treeCreator<HADRON>` to fill TTrees.
 - You register many histograms that are not filled. Are those needed?
 - You register histograms in `MC/reconstructed/`, while you fill them as `MC/reconstructed/background`
 - Check for typos: genearated -> generated; bakground -> background
 - Please use variables when filling histograms when you use it many times, e.g. `candidate.pt()`

**Comment by phymanshu** [comment]
Hi @lvermunt,
thanks for review.
* Yes, maybe this is not the convention. Earlier, I didn't find the `treeCreator` for `Xic`, so as a temporary solution I added the tree to the same task, but, I will create a `taskCreator` soon.
* The histograms for `Data`: As I see, they are filled when I test locally.
* The histograms for `MC`: I didn't touch them, those were there from before but I will correct those mistakes.
I will come back to you after the changes.

**Review by vkucera — CHANGES_REQUESTED** [review]
Hi @phymanshu , tasks are not supposed to fill trees. If you need a tree output, you should create a dedicated workflow that creates derived tables.

**Comment by phymanshu** [comment]
Hi @lvermunt and @vkucera 
I've made a change to the code.
* TTree is removed; I will make a separate PR for `treeCreatorXicToPKPi` later.
* Typos are corrected. @lvermunt, Indeed, the typos were related to our development. I didn't realize it before.

**Review by lvermunt — APPROVED** [review]

File: PWGHF/D2H/Tasks/taskXic.cxx [hunk 1]
```diff
@@ -31,175 +31,277 @@ using namespace o2::analysis::hf_cuts_xic_to_p_k_pi;
 #include "Framework/runDataProcessing.h"
 
 /// Ξc± analysis task
+
 struct HfTaskXic {
+
```
Comment (vkucera) [inline PWGHF/D2H/Tasks/taskXic.cxx:36]: Revert

File: PWGHF/D2H/Tasks/taskXic.cxx [hunk 2]
```diff
@@ -31,175 +31,277 @@ using namespace o2::analysis::hf_cuts_xic_to_p_k_pi;
...
+    registry.add("MC/reconstructed/background/hEtaRecBg", "3-prong candidates;candidate #it{#eta};;entries", {HistType::kTH2F, {{100, -2., 2.}, {vbins, "#it{p}_{T} (GeV/#it{c})"}}});
+    registry.add("MC/reconstructed/background/hPtProng0RecBg", "3-prong candidates;prong 0 #it{p}_{T} (GeV/#it{c});;entries", {HistType::kTH2F, {{100, 0., 10.}, {vbins, "#it{p}_{T} (GeV/#it{c})"}}});
+    registry.add("MC/reconstructed/background/hPtProng1RecBg", "3-prong candidates;prong 1 #it{p}_{T} (GeV/#it{c});;entries", {HistType::kTH2F, {{100, 0., 10.}, {vbins, "#it{p}_{T} (GeV/#it{c})"}}});
+    registry.add("MC/reconstructed/background/hPtProng2RecBg", "3-prong candidates;prong 2 #it{p}_{T} (GeV/#it{c});;entries", {HistType::kTH2F, {{100, 0., 10.}, {vbins, "#it{p}_{T} (GeV/#it{c})"}}});
   }
 
-  void process(soa::Join<aod::HfCand3Prong, aod::HfSelXicToPKPi> const& candidates)
+  void process(const o2::aod::Collision& collision, const soa::Join<aod::Tracks, aod::TracksDCA>& tracks, soa::Filtered<soa::Join<aod::HfCand3Prong, aod::HfSelXicToPKPi>> const& candidates, aod::BigTracksPID const&)
   {
```
Comment (vkucera) [inline PWGHF/D2H/Tasks/taskXic.cxx:143]:
```suggestion
  void process(aod::Collision const& collision, soa::Join<aod::Tracks, aod::TracksDCA> const& tracks, soa::Filtered<soa::Join<aod::HfCand3Prong, aod::HfSelXicToPKPi>> const& candidates, aod::BigTracksPID const&)
  {
```

File: PWGHF/D2H/Tasks/taskXic.cxx [hunk 3]
```diff
@@ -31,175 +31,277 @@ using namespace o2::analysis::hf_cuts_xic_to_p_k_pi;
...
-    for (auto& candidate : selectedXicCandidates) {
+    int nTracks = 0;
+
+    if (collision.numContrib() > 1) {
+      for (auto const& track : tracks) {
+        if (std::abs(track.eta()) > 4.0) { // hard coded cut TO BE REFINED
+          continue;
+        }
+        if (std::abs(track.dcaXY()) > 0.0025 || std::abs(track.dcaZ()) > 0.0025) { // hardcoded cut TO BE REFINED
```
Comment (vkucera) [inline PWGHF/D2H/Tasks/taskXic.cxx:151]: No magic numbers

File: PWGHF/D2H/Tasks/taskXic.cxx [hunk 4]
```diff
@@ -31,175 +31,277 @@ using namespace o2::analysis::hf_cuts_xic_to_p_k_pi;
...
-      registry.fill(HIST("hSelectionStatus"), candidate.isSelXicToPKPi(), candidate.pt());
-      registry.fill(HIST("hSelectionStatus"), candidate.isSelXicToPiKP(), candidate.pt());
-      registry.fill(HIST("hImpParErr"), candidate.errorImpactParameter0(), candidate.pt());
-      registry.fill(HIST("hImpParErr"), candidate.errorImpactParameter1(), candidate.pt());
-      registry.fill(HIST("hImpParErr"), candidate.errorImpactParameter2(), candidate.pt());
-      registry.fill(HIST("hDecLenErr"), candidate.errorDecayLength(), candidate.pt());
-      registry.fill(HIST("hDecLenErr"), candidate.chi2PCA(), candidate.pt());
+      auto prong0_P = trackProng0.p();
+      auto prong1_P = trackProng1.p();
+      auto prong2_P = trackProng2.p();
```
Comment (vkucera) [inline PWGHF/D2H/Tasks/taskXic.cxx:204]: No underscores in names of variables.

File: PWGHF/D2H/Tasks/taskXic.cxx [hunk 5]
```diff
@@ -31,175 +31,277 @@ using namespace o2::analysis::hf_cuts_xic_to_p_k_pi;
...
-  void processMc(soa::Join<aod::HfCand3Prong, aod::HfSelXicToPKPi, aod::HfCand3ProngMcRec> const& candidates,
-                 soa::Join<aod::McParticles, aod::HfCand3ProngMcGen> const& particlesMC, aod::BigTracksMC const&)
+  // Fill MC histograms
+  void processMc(soa::Filtered<soa::Join<aod::HfCand3Prong, aod::HfSelXicToPKPi, aod::HfCand3ProngMcRec>> const& candidates,
+                 soa::Join<aod::McParticles, aod::HfCand3ProngMcGen> const& particlesMC, aod::BigTracksMC const& /*tracks*/)
   {
+
     // MC rec.
-    for (auto& candidate : selectedMCXicCandidates) {
+    for (auto& candidate : candidates) {
```
Comment (vkucera) [inline PWGHF/D2H/Tasks/taskXic.cxx:239]:
```suggestion
    for (auto const& candidate : candidates) {
```

File: PWGHF/D2H/Tasks/taskXic.cxx [hunk 6]
```diff
@@ -31,175 +31,277 @@ using namespace o2::analysis::hf_cuts_xic_to_p_k_pi;
...
       auto mass = 0.;
       auto massC = 0.;
+      auto candidatePt = candidate.pt();
 
       if (candidate.isSelXicToPKPi() >= selectionFlagXic) {
         mass = invMassXicToPKPi(candidate);
       }
       if (candidate.isSelXicToPiKP() >= selectionFlagXic) {
-        massC = invMassXicToPiKP(candidate);
+        massC = invMassXicToPiKP(candidate); // mass conjiugated
```
Comment (vkucera) [inline PWGHF/D2H/Tasks/taskXic.cxx:256]: The names `mass` and `massC` are not clear enough. Please rename to something better.

**Review by vkucera — CHANGES_REQUESTED** [review]

**Comment by lvermunt** [comment]
This PR can be closed right @phymanshu?

**Comment by phymanshu** [comment]
Closing this PR since this is no longer relevant. A new PR has been made.

**Pull request closed without merging**